components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityTenantUtil.java

```diff
@@ -188,16 +194,21 @@ public static void setRealmService(RealmService realmService) {
         IdentityTenantUtil.realmService = realmService;
     }
 
-    public static void setTenantRegistryLoader(TenantRegistryLoader tenantRegistryLoader) {
-        IdentityTenantUtil.tenantRegistryLoader = tenantRegistryLoader;
-    }
-
-    public static void setBundleContext(BundleContext bundleContext) {
-        IdentityTenantUtil.bundleContext = bundleContext;
+    /**
+     * @deprecated
+     *
+     * This method will be removed in the upcoming major release.
+     * Because, tenant domain can be retrieved using tenantId.
+     * Use {@link #initializeRegistry(int)} instead.
+     *
+     */
+    @Deprecated
+    public static void initializeRegistry(int tenantId, String tenantDomain) throws IdentityException {
+        initializeRegistry(tenantId);
     }
 
-
-    public static void initializeRegistry(int tenantId, String tenantDomain) throws IdentityException {
+    public static void initializeRegistry(int tenantId) throws IdentityException {
```

Comment:
Let's not modify the method signature of an existing public method. Instead, add an overloaded method and call it from within the old method